File: Gheetah/Templates/C#/Base_Desktop/Drivers/DriverFactory.cs
using OpenQA.Selenium.Appium;
using OpenQA.Selenium.Appium.Windows;

namespace {{ProjectName}}.Drivers
{
    public static class DriverFactory
    {
        public static WindowsDriver CreateDriver(string appPath, string deviceName = "WindowsPC")
        {
            var options = new AppiumOptions();
            options.AddAdditionalAppiumOption("app", appPath);
            options.AddAdditionalAppiumOption("deviceName", deviceName);
            options.AddAdditionalAppiumOption("platformName", "Windows");
            options.AddAdditionalAppiumOption("ms:waitForAppLaunch", "10");

            var driver = new WindowsDriver(new Uri("http://127.0.0.1:4723"), options);
            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);
            return driver;
        }
    }
}

File: Gheetah/Templates/Java/Base_Desktop/src/main/java/{{ProjectName}}/drivers/DriverFactory.java
package {{ProjectName}}.drivers;

import io.appium.java_client.windows.WindowsDriver;
import org.openqa.selenium.remote.DesiredCapabilities;
import java.net.URL;
import java.time.Duration;

public class DriverFactory {
    private static WindowsDriver driver;

    public static WindowsDriver createDriver(String appPathOrId) throws Exception {
        DesiredCapabilities caps = new DesiredCapabilities();
        // appPathOrId: .exe yolu veya Windows AppId (örn: Microsoft.WindowsCalculator_8wekyb3d8bbwe!App)
        caps.setCapability("app", appPathOrId);
        caps.setCapability("platformName", "Windows");
        caps.setCapability("deviceName", "WindowsPC");
        caps.setCapability("ms:waitForAppLaunch", "5");

        // WinAppDriver varsayılan olarak 4723 portunda çalışır
        URL url = new URL("http://127.0.0.1:4723/");
        
        driver = new WindowsDriver(url, caps);
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        return driver;
    }

    public static WindowsDriver getDriver() { return driver; }

    public static void quitDriver() {
        if (driver != null) {
            driver.quit();
            driver = null;
        }
    }
}